misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/model/BallerinaOpenApi.java

```diff
@@ -127,8 +127,37 @@ private void setPaths(OpenAPI openAPI) throws BallerinaOpenApiException {
                     }
                 });
             }
-            paths.add(new AbstractMap.SimpleEntry<>(path.getKey(), balPath));
+            String resourcePath;
+            // Use ballerina string templating if there are path params.
+            if (hasPathParams(path.getValue())) {
+                resourcePath = "string `" + path.getKey().replace("{", "${") + "`";
+            } else {
+                resourcePath = "\"" + path.getKey() + "\"";
+            }
+            paths.add(new AbstractMap.SimpleEntry<>(resourcePath, balPath));
+        }
+    }
+
+    /**
+     * Checks if there are any path parameters in the operations.
+     * @param path  The path item in the OpenAPI spec.
+     * @return True if there are path parameters, else false.
+     */
+    private boolean hasPathParams(PathItem path) {
+        if (path.getParameters() != null && path.getParameters().size() > 0) {
+            return path.getParameters().stream()
+                    .anyMatch(parameter -> parameter.getIn() != null && parameter.getIn().equals("path"));
```

Comment:
Shall we assign the string `"path" ` to private static final variable and use it here instead of hard coding the value directly?
`private static final String PATH = "path"`